lib/fog/azurerm/models/network/network_interface.rb

```diff
@@ -61,13 +68,57 @@ def save
           requires :subnet_id
           requires :ip_configuration_name
           requires :private_ip_allocation_method
-          network_interface = service.create_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method)
+          network_interface = service.create_or_update_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method, private_ip_address)
+          merge_attributes(Fog::Network::AzureRM::NetworkInterface.parse(network_interface))
+        end
+
+        def update(updated_attributes = {})
+          validate!(updated_attributes)
+          merge_attributes(updated_attributes)
+          network_interface = service.create_or_update_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method, private_ip_address)
+          merge_attributes(Fog::Network::AzureRM::NetworkInterface.parse(network_interface))
+        end
+
+        def attach_subnet(subnet_id)
+          raise 'Subnet ID can not be nil.' if subnet_id.nil?
+          network_interface = service.attach_resource_to_nic(resource_group, name, 'Subnet', subnet_id)
+          merge_attributes(Fog::Network::AzureRM::NetworkInterface.parse(network_interface))
+        end
+
+        def attach_public_ip(public_ip_id)
+          raise 'Public-IP ID can not be nil.' if public_ip_id.nil?
+          network_interface = service.attach_resource_to_nic(resource_group, name, 'Public-IP-Address', public_ip_id)
```

Comment:
Define 'Subnet', 'Public-IP-Address' and others as constants in file beginning. Use them here instead